### DISCUSSION TITLE
[QUESTION] How to integrate it into a pure CPP environment

### DISCUSSION THREAD

**Discussion by tolstenko** [trustwallet/wallet-core#2012] [Q&A]
**Is your feature request related to a problem? Please describe.**
I am new to this project and I feel it lacks a more detailed CPP example.

**Describe the solution you'd like**
A better CPP example containing
- How to submit the raw transaction to an RPC (curl? cpr?)
- How to estimate gas from a transaction
- Call contract function using the ABI and Bytecode

**Describe alternatives you've considered**
- https://github.com/ethereumproject/cpp-ethereum is deprecated
- I came from ethersjs, so obviously I am biased, but for now, the easiest example I found to do what I need in CPP is yours.  https://github.com/trustwallet/wallet-core/blob/master/samples/cpp/sample.cpp 
- The options I had in mind is to create a side-car or even embed nodejs(ethers) or go(geth) into CPP in order to achieve what I need but it seems too naive.

**Additional context**
- I am developing a game in CPP and I am looking for solutions to embed a wallet and execute contract transactions inside the game. The easiest example so far in CPP is yours.

Reference:
- You can see here, CPP doesn't have any maintained client support. https://ethereum.org/en/developers/docs/nodes-and-clients/#execution-clients
![image](https://user-images.githubusercontent.com/2386417/153590283-9d335b0d-f1da-4f6b-903f-81a48890b28e.png)

Thanks for your time.

**Comment by optout21** [comment]
Communication to RPC nodes is outside of the scope of wallet-core, though of course it is an important part of a full product.
Broadcasting a prepared Eth transaction to an RPC node is quite easy, a simple google search and a curl command can do it.  In fact one does not even need a curl command, etherscan has a built-in function to broadcast a transaction.
